core/base/integralLines/IntegralLines.h

```diff
@@ -72,9 +173,157 @@ namespace ttk {
       direction_ = direction;
     }
 
-    int preconditionTriangulation(
-      ttk::AbstractTriangulation *triangulation) const {
-      return triangulation->preconditionVertexNeighbors();
+    void findNextVertex(ttk::SimplexId &vnext,
+                        ttk::SimplexId &fnext,
+                        std::vector<ttk::SimplexId> &component,
+                        const SimplexId *offsets) const;
+
+#ifdef TTK_ENABLE_MPI
+
+    /**
+     * @brief Constructs the global identifiers for vertices and edges.
+     *
+     * @tparam triangulationType
+     * @param globalVertexId vector of global identifiers for vertices
+     * @param globalCellId vector of global identifiers for cells
+     * @param seedIdentifiers linked list of identifiers of seed, one for each
+     * integral line
+     * @param forkIdentifiers linked list of identifiers of forks, one for each
+     * integral line
+     * @param trajectories linked list of identifiers of forks, one for each
+     * integral line
+     * @param localVertexIdentifiers linked list of identifiers, one for each
+     * vertex of an integral lines. It represents the number of the vertex in
+     * the integral line.
+     * @param triangulation Triangulation
+     * @return int
+     */
+    template <typename triangulationType>
+    int getGlobalIdentifiers(
+      std::vector<ttk::SimplexId> &globalVertexId,
+      std::vector<ttk::SimplexId> &globalCellId,
+      std::vector<ttk::ArrayLinkedList<ttk::SimplexId, TABULAR_SIZE>>
+        &seedIdentifiers,
+      std::vector<ttk::ArrayLinkedList<ttk::SimplexId, TABULAR_SIZE>>
+        &forkIdentifiers,
+      std::vector<ttk::ArrayLinkedList<std::vector<ttk::SimplexId>,
+                                       TABULAR_SIZE>> &trajectories,
+      std::vector<ttk::ArrayLinkedList<std::vector<ttk::SimplexId>,
+                                       TABULAR_SIZE>> &localVertexIdentifiers,
+      triangulationType *triangulation);
+
+    /**
+     * @brief Sorts ghosts and exchanges their global identifiers between
+     * processes.
+     *
+     * @param unmatchedGhosts Ghosts vertices and edges for which the global
+     * identifier is to be determined
+     * @param globalVertexId vector of global identifiers for vertices
+     * @param globalCellId vector of global identifiers for cells
+     * @return int 0 for success
+     */
+    int exchangeGhosts(std::vector<std::vector<ttk::intgl::GhostElementsToSort>>
```

Comment:
can't that function (or significant parts of it) go to `core/base/common/MPIUtils.h`?

Reply:
Everything in this function is very specific to integral lines. The parameters used here to identify each segment of integral line are not used by any other filter, therefore keeping the code in the integral lines filter is better, as it will never be called by any other filter.